generators/base/generator-base.mjs

```diff
@@ -203,12 +206,7 @@ export default class JHipsterBaseGenerator extends PrivateBase {
    * @returns {string}
    */
   jhipsterTemplatePath(...args) {
```

Comment:
IMO the implementation should be keep here.

Reply:
I really want to move every single line of logic in dedicated file to see how it can be reused over the entire framework. I also want to strictly follow SOLID principles all over the business logic to maximize reuse.
The generator class (and all non-business logic related classes) will act as facade over this well OOP structured core logic files

Reply:
I cannot see another use of this extracted function.

Reply:
Can you think about any use of this?

Reply:
Yes...

The logic could be enhance to support when testing an home-made blueprint (part of the path is local, wether most of the paths are in the generator-jhipster package.

Reply:
I don't get why your are resolving questioning I am creating.
The generator is a class with inheritance, splitting into small utilities won't simplify the implementation.
I won't approved this.
Methods that belongs to the generator-base should be kept here, unless it's very complicated.

Reply:
Let take the example of code formatting: that are 6 different methods, each of them trimming, replacing specific characters, indenting, with a bunch of duplicated things. Suppose that we should change the behavior now: we'll modify all these methods, without knowing where they are, with the according amount of bugs, issues, etc.
Centralizing logic will help fixing this kind of issues.

Also, the current codebase is far from following the best code conventions (i.e. assignements + logic) and it's mostly due to this monolithic 2K+ lines of classes.

There are a lot of errors done with not centralizing these small methods somewhere.

On this specific use lines:
From a strict SOLID point of view, this method basically does two things: computing the root generator path, then fetching the path of a template.
While we for sure will only have one use case for locating template, we will for sure have many different kind of use case needing to get the root of the generator.
I'm also sure that it will help with typescript migration, feature addition, ...

Please don't be against this feature: I'm convinced that this will bring some new dimension to the framework, and most of the committers are for it, [at least deepu, sendil, pierre and me](https://groups.google.com/g/jhipster-dev/c/M1IKwib7i08/m/vV8DDnQ7AAAJ) and
https://groups.google.com/g/jhipster-dev/c/y_sER_CM3Hc/m/MhpXR3DaBAAJ

Reply:
Don't get me wrong, I am not denying any feature.
We probably have more than 100 methods at generator-base/private to extract.
`jhipsterTemplatePath`, `removeFolder`, `removeFile` implementation should be kept at the generator-base, at least for now.
From time to time I glance to see what to remove or move around, if the implementation is not local, I will have to open the other function.

Reply:
We could always mark them as private in the jsdoc, can't we?

Reply:
rolledback, but not convinced